controllers/global_search_setup.go

```diff
@@ -0,0 +1,558 @@
+				component.(map[string]interface{})["enabled"] == true {
+				log.V(5).Info("Cluster Proxy add-on is enabled.")
+				clusterProxyEnabled = true
+			}
+		}
+		if !managedServiceAccountEnabled {
+			return fmt.Errorf("Managed Service Account add-on is not enabled in MulticlusterEngine.")
+		}
+		if !clusterProxyEnabled {
+			return fmt.Errorf("Cluster Proxy add-on is not enabled in MulticlusterEngine.")
+		}
+	}
+	return nil
+}
+
+// Logic to enable Global Search.
+//  1. Enable global search feature in the console.
+//     a. Add globalSearchFeatureFlag=true to configmap console-mce-config in multicluster-engine namespace.
+//     b. Add globalSearchFeatureFlag=true to configmap console-config in open-cluster-management namespace.
+//  2. Enable federated search feature in the search-api deployment.
+//  3. Create configuration resources for each Managed Hub.
+//     a. Create a ManagedServiceAccount search-global.
+//     b. Create a ManifestWork search-global-config if it doesn't exist.
+func (r *SearchReconciler) enableGlobalSearch(ctx context.Context, instance *searchv1alpha1.Search) error {
+	// 1. Enable global search feature in the console.
+	// 1a.Add globalSearchFeatureFlag=true to configmap console-mce-config in multicluster-engine namespace.
+	err := r.updateConsoleConfig(ctx, true, "multicluster-engine", "console-mce-config")
+	if err != nil {
+		log.Error(err, "Failed to enable the global search feature in console-mce-config.")
+		return err // QUESTION: Should we return here or continue? Should we rollback other changes?
+	}
+	// 1b. Add globalSearchFeatureFlag=true to configmap console-config in open-cluster-management namespace.
+	err = r.updateConsoleConfig(ctx, true, instance.GetNamespace(), "console-config")
+	if err != nil {
+		log.Error(err, "Failed to enable the global search feature in console-config.")
+		return err // QUESTION: Should we return here or continue? Should we rollback other changes?
+	}
+
+	// 2. Enable federated search feature in the search-api deployment.
+	err = r.updateSearchApiDeployment(ctx, true, instance)
+	if err != nil {
+		log.Error(err, "Failed to enable the federated global search feature on search-api deployment.")
+		return err // QUESTION: Should we return here or continue? Should we rollback other changes?
+	}
+
+	// Create configuration resources for each Managed Hub.
+	// MANAGED_HUBS=($(oc get managedcluster -o json | jq -r '.items[] | select(.status.clusterClaims[] |
+	//     .name == "hub.open-cluster-management.io" and .value != "NotInstalled") | .metadata.name'))
+	clusterList, err := r.DynamicClient.Resource(managedClusterResourceGvr).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		log.Error(err, "Failed to list the ManagedClusters to configure global search.")
+		return err // QUESTION: Should we return here or continue? Should we rollback other changes?
+	}
+
+	for _, cluster := range clusterList.Items {
```

Comment:
Put this in `else` or check `if  clusterList is not nil` and items are present? Wondering if it will go into 'nilPointer' error otherwise.